File: TileMapCreator/tile_palette.py
# tile_palette.py
from PySide6.QtWidgets import QWidget, QVBoxLayout, QLabel, QScrollArea
from PySide6.QtGui import QPixmap, QMouseEvent
from PySide6.QtCore import Qt, Signal

class TilePalette(QWidget):
    tile_selected_sig = Signal(QPixmap)

    def __init__(self):
        super().__init__()
        self.layout = QVBoxLayout()
        self.setLayout(self.layout)

        self.tiles = []

        self.update_tilepalette()


    def update_tilepalette(self):
        for t in self.tiles:
            self.layout.addWidget(t)

    def select_tile(self, pixmap):
        self.tile_selected_sig.emit(pixmap)

    def add_tiles(self, tile_images):
        for t in tile_images:
            tile = ClickableTile(t)
            tile.clicked_sig.connect(self.select_tile)
            self.tiles.append(tile)
        self.update_tilepalette()
            


class ClickableTile(QLabel):
    clicked_sig = Signal(QPixmap)

    def __init__(self, pixmap):
        super().__init__()
        self.setPixmap(pixmap)
        self.setFixedSize(pixmap.size())

    def mousePressEvent(self, event: QMouseEvent):
        if event.button() == Qt.LeftButton:
            self.clicked_sig.emit(self.pixmap())

File: TileMapCreator/CppRewrite/Sources/TilePalette.cpp
#include "TilePalette.h"
#include "ClickableTile.h"
#include <QWidget>
#include <qpixmap.h>
#include <spdlog/spdlog.h>

TilePalette::TilePalette() : layout() {
    spdlog::debug("creating palette");
    setLayout(&layout);

    update_tilepalette();
}

void TilePalette::update_tilepalette() {
    for (int i = 0; i < tiles.size(); i++) {
        layout.addWidget(tiles[i].get());
    }
}

void TilePalette::select_tile(QPixmap pixmap) {
    emit tile_selected_sig(pixmap);
}

void TilePalette::add_tiles(QList<std::shared_ptr<QPixmap>> tile_images) {

    spdlog::debug("Debut de add tiles");
    for (int i = 0; i < tile_images.size(); i++) {
        spdlog::debug("Ajout d'une tile...");
        tiles.push_back(
            std::make_shared<ClickableTile>(*(tile_images[i].get())));
        connect(tiles[tiles.size() - 1].get(), &ClickableTile::clicked_sig,
                this, &TilePalette::select_tile);
    }
    spdlog::debug("Updating tile palette !");
    update_tilepalette();
}